lib/open_trip_planner_client/itinerary_group.ex

```diff
@@ -39,11 +43,49 @@ defmodule OpenTripPlannerClient.ItineraryGroup do
   """
   @spec groups_from_itineraries([Itinerary.t()], Keyword.t()) :: [%__MODULE__{}]
   def groups_from_itineraries(itineraries, opts \\ []) do
+    ideal_itineraries = opts |> Keyword.get(:ideal_itineraries, [])
+
+    available_groups =
+      itineraries
+      |> to_groups(opts)
+      |> Enum.take(Keyword.get(opts, :num_groups, @num_groups))
+
+    unavailable_groups =
+      ideal_itineraries
+      |> filter_unavailable_itineraries(available_groups)
+      |> to_groups(opts |> Keyword.put(:available?, false))
+
+    unavailable_groups ++ available_groups
+  end
+
+  defp filter_unavailable_itineraries(ideal_itineraries, available_groups) do
+    cost_threshold =
+      available_groups
+      |> Stream.map(&generalized_cost/1)
+      |> Enum.min()
+
+    available_identifiers = available_groups |> MapSet.new(& &1.identifier)
+
+    ideal_itineraries
+    |> Enum.filter(fn itinerary ->
+      unavailable? = !MapSet.member?(available_identifiers, Itinerary.group_identifier(itinerary))
+      cost_below_threshold? = itinerary.generalized_cost <= cost_threshold
+      has_tag? = !is_nil(itinerary[:tag])
+
+      unavailable? && (cost_below_threshold? || has_tag?)
```

Comment:
suggestion: maybe we shouldn't consider whether the ideal itinerary has a tag here? we've seen a few results where the tagged result is pretty lousy (e.g. a "direct" trip that involves a long walk).

Reply:
Let's think about this over the weekend... 

My rationale for this was that if a trip got tagged, then that would cause it to show up under normal circumstances, and the whole point of this is to flag that "yes, we know this trip usually shows up, but it's not right now for good reason".

If we're questioning the wisdom of the tags, maybe that means we should revisit the whole concept of tagging? And/or the specific way that we're tagging?

Reply:
We ended up not tagging the unavailable trips (and leaving the current tagging as is)!

Reply:
Correction: we're not _evaulating_ tags in comparing with available trips